pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1160,17 +1183,22 @@ func (p *PriorityQueue) podMatchesEvent(podInfo *framework.QueuedPodInfo, cluste
 
 		// Secondly verify the plugin name matches.
 		// Note that if it doesn't match, we shouldn't continue to search.
-		if evtMatch && intersect(nameSet, podInfo.UnschedulablePlugins) {
+		if evtMatch && intersect(plugins, podInfo.UnschedulablePlugins) {
```

Comment:
I find it odd that we are not checking the hint here.

Basically, the event matching is split in two places: here and the eventhandlers code. Should it all be moved here?
Although, maybe worth in a follow up PR, as this one is already long.

Reply:
similar with my previous comment https://github.com/kubernetes/kubernetes/pull/117561#issuecomment-1536655434. It's cost effective to check if events match first, and then run hint functions.

Reply:
> could you evaluate if moving the logic of:
> 
> ```go
> 	if len(pInfo.UnschedulablePlugins) != 0 && !p.podMatchesEvent(pInfo, event) {
> 		return false
> 	}
> ```
> 
> ahead of calling hintFns can be more performant?

I benchmarked such a change and while it didn't make much difference with just a single hint function, it was still a bit better. It should become more important when adding more hint functions. See the new `eventMightMakePodSchedulable` in the latest commit.

Reply:
Regarding `podInfo.UnschedulablePlugins`: I wonder whether it is still needed when more plugins provide hint functions. But exploring that would make this PR even larger, so let's not do this now.

Interestingly, @sanposhiho is suggesting the opposite in https://github.com/kubernetes/kubernetes/pull/117844#discussion_r1189844110: for `Requeue` to make the right decision regarding concurrent cluster events, all failed plugins must be listed in `UnschedulablePlugins`. Currently they are not, so he is suggesting to add them also when they fail in other extension points, like `Reserve`.

Reply:
My initial motivation to start this proposal (#114297) is "to ask the plugins which caused the scheduling failure about whether events make this Pod schedulable or not".
So, calling functions (Requeue or EnqueueExtensionsWithHints) of only UnschedulablePlugins; this is must-to-have.
If PodA is failed by PodAffinity, then we want to hear the opinion **_only_** from PodAffinity plugin to decide whether put this Pod to activeQ/backoffQ by this event.

As talking in https://github.com/kubernetes/kubernetes/pull/117844#discussion_r1189844110, I agree that we should improve which plugin to register to unschedulable plugins though. For @pohly's usecase, we should add plugin failed in Reserve at least.